dask/array/reductions.py

```diff
@@ -326,17 +326,15 @@ def partial_reduce(
 
 @wraps(chunk.sum)
 def sum(a, axis=None, dtype=None, keepdims=False, split_every=None, out=None):
-    if dtype is not None:
-        dt = dtype
-    else:
-        dt = getattr(np.empty((1,), dtype=a.dtype).sum(), "dtype", object)
+    if dtype is None:
+        dtype = getattr(np.zeros(1, dtype=a.dtype).sum(), "dtype", object)
```

Comment:
Fixes random crash caused by applying sum() on random bytes.
``np.empty(0, )`` would have been nicer but fails on very old versions of numpy.